src/Document/Editor.js

```diff
@@ -0,0 +1,123 @@
+//import FindTitle from "./FindTitle.js"; 구현 실패
+import { request } from "../../library/api.js";
+import { changeRootTitle } from "../../library/titleChanger.js";
+import { markdownToHtml, htmlToMarkdown } from "../../library/convert.js";
+export default function Editor({ $page, initialState }) {
+  this.state = initialState;
+
+  const handleKeyUp = async (e) => {
+    if (e.target.className === "editor-title") {
+      handleTitle(e.target.value);
+    } else if (e.target.className === "editor-content") {
+      handleContent(e.target);
+    }
+  };
+
+  const handleTitle = async (e) => {
+    const nextState = {
+      ...this.state,
+      title: e,
+    };
+    if (e === "") {
+      nextState.title = "제목없음";
+    }
+    this.setState(nextState);
+    await fetchEditDoc(this.state);
+    changeRootTitle(this.state);
+  };
+
+  const handleContent = async (e) => {
+    const content = markdownToHtml(e.innerText);
+    const nextState = {
+      ...this.state,
+      content: content,
+    };
+    this.setState(nextState);
+    //findTitle.setState(nextState); Find Title 실패
+    //await findTitle.findMatchedTitle(nextState.content);
+    await fetchEditDoc(this.state);
+  };
+
+  //const findTitle = new FindTitle({ $page, initialState }); 실패
+  const fetchEditDoc = async (document) => {
+    await request(`/documents/${document.id}`, {
+      method: "PUT",
+      body: JSON.stringify({
+        title: `${document.title}`,
+        content: `${document.content}`,
+      }),
+    });
+  };
```

Comment:
`document`가 객체면 아래처럼 parameter를 받아서 사용해보셔도 좋을거 같습니다 ㅎㅎ

```suggestion
  const fetchEditDoc = async ({ id, title, content }) => {
    await request(`/documents/${id}`, {
      method: "PUT",
      body: JSON.stringify({
        title: `${title}`,
        content: `${content}`,
      }),
    });
  };
```